tests/integration/init/test_init_command.py

```diff
@@ -147,12 +151,48 @@ def test_init_command_with_extra_context_parameter(self):
                 raise
 
             self.assertEqual(process.returncode, 0)
-            self.assertTrue(os.path.isdir(temp + "/sam-app-maven"))
+            self.assertTrue(Path(temp, "sam-app-maven").is_dir())
+
+
+class TestInitWithArbitraryProject(TestCase):
+    def setUp(self):
+        self.tempdir = tempfile.mkdtemp()
+
+        zipdata_folder = Path(self.tempdir, "zipdata")
+        zipdata_folder.mkdir(parents=True)
+        Path(zipdata_folder, "test.txt").write_text("hello world")
+
+        zip_path_no_extension = str(Path(self.tempdir, "myfile"))
+
+        self.zip_path = shutil.make_archive(zip_path_no_extension, "zip", root_dir=self.tempdir, base_dir="zipdata")
+
+    def tearDown(self):
+        shutil.rmtree(self.tempdir)
+
+    @parameterized.expand([(None,), ("project_name",)])
+    def test_arbitrary_project(self, project_name):
+        with tempfile.TemporaryDirectory() as temp:
```

Comment:
are there any gotchas with temporary directories on windows?

Reply:
it didn't break before ;) so no